File: ios/Classes/FlutterSocialSharePlugin.swift
import Flutter
import UIKit
import FBSDKShareKit
import FBSDKCoreKit

public class FlutterSocialSharePlugin: NSObject, FlutterPlugin {
    private var facebookShareHandler: FacebookShareHandler?
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "flutter_social_share", binaryMessenger: registrar.messenger())
        let instance = FlutterSocialSharePlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    deinit {
        // Ensure cleanup when plugin is deallocated
        facebookShareHandler?.cleanup()
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformVersion":
            result("iOS " + UIDevice.current.systemVersion)
        case "initializeFacebook":
            handleFacebookInitialize(call: call, result: result)
        case "shareImageToFacebook":
            handleFacebookShareImage(call: call, result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
    
    private func handleFacebookInitialize(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let arguments = call.arguments as? [String: Any],
              let appId = arguments["appId"] as? String,
              let clientToken = arguments["clientToken"] as? String else {
            result(FlutterError(code: "INVALID_ARGUMENTS", 
                              message: "App ID and Client Token are required", 
                              details: nil))
            return
        }
        
        facebookShareHandler = FacebookShareHandler()
        facebookShareHandler?.initialize(appId: appId, clientToken: clientToken, result: result)
    }
    
    private func handleFacebookShareImage(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let arguments = call.arguments as? [String: Any],
              let imagePath = arguments["imagePath"] as? String else {
            result(FlutterError(code: "INVALID_ARGUMENTS", 
                              message: "Image path is required", 
                              details: nil))
            return
        }
        
        let caption = arguments["caption"] as? String
        
        guard let shareHandler = facebookShareHandler else {
            result(FlutterError(code: "NOT_INITIALIZED", 
                              message: "Facebook SDK not initialized", 
                              details: nil))
            return
        }
        
        shareHandler.shareImage(imagePath: imagePath, caption: caption, result: result)
    }
}

File: android/src/main/kotlin/com/dovireinfotech/flutter_social_share/FlutterSocialSharePlugin.kt
package com.dovireinfotech.flutter_social_share

import android.app.Activity
import android.content.Context
import android.content.Intent
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.plugin.common.PluginRegistry

/** FlutterSocialSharePlugin */
class FlutterSocialSharePlugin :
    FlutterPlugin,
    MethodCallHandler,
    ActivityAware,
    PluginRegistry.ActivityResultListener {
    
    // The MethodChannel that will handle communication between Flutter and native Android
    private lateinit var channel: MethodChannel
    private var context: Context? = null
    private var activity: Activity? = null
    private var facebookShareHandler: FacebookShareHandler? = null

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        context = flutterPluginBinding.applicationContext
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_social_share")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(
        call: MethodCall,
        result: Result
    ) {
        when (call.method) {
            "getPlatformVersion" -> {
                result.success("Android ${android.os.Build.VERSION.RELEASE}")
            }
            "initializeFacebook" -> {
                handleFacebookInitialize(call, result)
            }
            "shareImageToFacebook" -> {
                handleFacebookShareImage(call, result)
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    private fun handleFacebookInitialize(call: MethodCall, result: Result) {
        val appId = call.argument<String>("appId")
        val clientToken = call.argument<String>("clientToken")
        
        if (appId == null || clientToken == null) {
            result.error("INVALID_ARGUMENTS", "App ID and Client Token are required", null)
            return
        }
        
        context?.let { ctx ->
            facebookShareHandler = FacebookShareHandler(ctx)
            facebookShareHandler?.initialize(appId, clientToken, result)
        } ?: result.error("NO_CONTEXT", "Application context not available", null)
    }

    private fun handleFacebookShareImage(call: MethodCall, result: Result) {
        val imagePath = call.argument<String>("imagePath")
        val caption = call.argument<String>("caption")
        
        if (imagePath == null) {
            result.error("INVALID_ARGUMENTS", "Image path is required", null)
            return
        }
        
        activity?.let { act ->
            facebookShareHandler?.shareImage(act, imagePath, caption, result)
        } ?: result.error("NO_ACTIVITY", "Activity not available for sharing", null)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
        context = null
        facebookShareHandler = null
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activity = binding.activity
        binding.addActivityResultListener(this)
    }

    override fun onDetachedFromActivityForConfigChanges() {
        activity = null
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        activity = binding.activity
        binding.addActivityResultListener(this)
    }

    override fun onDetachedFromActivity() {
        activity = null
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
        // Forward activity results to Facebook CallbackManager
        return FacebookCallbackManager.getInstance().onActivityResult(requestCode, resultCode, data)
    }
}
